enable_disable_persistence.md

```diff
@@ -0,0 +1,180 @@
+other service management tools (like
+[Edward](http://engblog.yext.com/edward/commands/#start)).
+
+`systemd` is a persistent daemon that runs admin-level services on linux.
+
+You use `systemctl start` and `systemctl stop` to start/stop services.
+
+And `systemctl enable` and `systemctl disable` to enable/disable services to
+auto-boot at system startup.
+
+Note that `enable` does not start a service immediately. In other words, systemd
+models "start on boot" and "start now" as independent actions.
+
+## Product Specification
+
+### Run all, Disable from UI
+
+1. Write a Tiltfile with services Paulie and Roberta
+2. Run `tilt up` -> Paulie and Roberta are enabled.
+4. Click "disable" on "Paulie" -> Roberta is enabled, Paulie is disabled
+5. Ctrl-c and re-run `tilt up` -> Roberta is enabled, Paulie is disabled
+
+Possible Principle:
+
+`tilt up` (without arguments) always inherits all the enable/disable overides
+from the previous Tilt session.
+
+### Specify services, Disable from UI
+
+1. Write a Tiltfile with services Paulie and Roberta and Joe
+2. Run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled.
+4. Click "disable" on "Paulie" -> Roberta is enabled, Joe and Paulie are disabled.
+5. Ctrl-c and re-run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled.
+
+Possible Principle:
+
+Tilt always respects arguments to `tilt up` when it starts.
+
+### Specify services, Enable from UI
+
+1. Write a Tiltfile with services Paulie and Roberta and Joe
+2. Run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled.
+4. Click "enable" on "Joe" -> Paulie and Roberta and Joe are enabled.
+5. Ctrl-c and re-run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled. 
+
+Possible Principle:
+
+Tilt always respects arguments to `tilt up` when it starts.
+
+Your UI change has been wiped out by the `tilt up` args.
+
+### Run all, Disable from UI, Change Args
+
+1. Write a Tiltfile with services Paulie and Roberta
+2. Run `tilt up` -> Paulie and Roberta are enabled.
+4. Click "disable" on "Paulie" -> Roberta is enabled, Paulie is disabled.
+5. Run `tilt args paulie roberta` in a separate terminal. -> Tiltfile re-executes, Paulie and Roberta are enabled. 
+
+Possible Principle:
+
+Changing the tilt args wipes out any enabled/disabled services from the UI.
+
+### Run one service, Enable from UI, Change Args
+
+1. Write a Tiltfile with services Paulie and Roberta
+2. Run `tilt up paulie` -> Paulie is enabled, Roberta is disabled.
+4. Click "enable" on "Roberta" -> Paulie and Roberts is enabled.
+5. Run `tilt args paulie` in a separate terminal. -> Tiltfile re-executes, Paulie and Roberta stay enabled.
+6. Run `tilt args roberta` in a separate terminal. -> Tiltfile re-executes, Roberta is enabled, Paulie is disabled.
+
+Possible Principle:
+
+Changing the tilt args wipes out any enabled/disabled services from the UI. But
+keeping the tilt args the same does nothing.
+
```

Comment:
I think we need some examples w/ `config.set_enabled_services` - these are the cases that are tripping me up and making me consider separating service selection from tilt args.
e.g.:
```
services = config.define_string_list('services', args=True)
cfg = config.parse()
services = ['roberta'] + cfg['services']
config.set_enabled_services(services)
```
and then:
1. `tilt up paulie` -> Roberta and Paulie are enabled
2. Disable Roberta from the UI.
3. Run `tilt up juliana`. The user specified 'juliana', but the `config.set_enabled_resources` call got 'juliana' and 'roberta'

A CLI that gets directly at services (or groups) and allows adding/removing rather than replacing seems like it simplifies this a lot. It seems worth looking into how much our users actually want to do arbitrarily complex things with args -> services vs just directly supporting the enabling/disabling of services and groups of services.

Reply:
sure, added a `set_enabled_resources` case, I'm not sure it makes the implementation any more complex but maybe i'm missing something